src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/datetimechecker/DateRelevantForCurrentReconciliationChecker.java

```diff
@@ -0,0 +1,16 @@
+package uk.gov.hmcts.reform.blobrouter.reconciliation.service.datetimechecker;
+
+import java.time.ZonedDateTime;
+
+/**
+ * Checker to ensure the provided DateTime is relevant to currently running reconciliation process.
+ *
+ * The DateTime will be relevant only if it's before SummaryReport generation time, i.e.:
+ * <ul>
+ * <li>If SummaryReport is generated every hour until 11:00 for yesterdays envelopes, and we ask if 10:45 yesterday is relevant the result is true</li>
+ * <li>If SummaryReport is generated every hour until 11:00 for yesterdays envelopes, and we ask if 11:15 yesterday is relevant the result is false</li>
+ * </ul>
+ */
+public interface DateRelevantForCurrentReconciliationChecker {
```

Comment:
do we need an interface?

Reply:
I was thinking the same, but decided yes, for flexibility. My thinking behind the interface is, we can even start with Dummy hardcoded version, checking time lesser than 6am (what Mustafa suggested...), or my simple hour crone aware checker, and then in future implement more serious checkers (including the one fully aware of the minutes and days). 
Of course we could just modify the same class... but I like the separation nicer in this case. 

I believe it will be easier for the tests and abstraction purposes, so I would defend the interface here - the controller wouldn't care what logic sits behind the warning, it just wants something which will return true or false.